### PR TITLE
Switch waiting mechanism in function calling HEPnOS thread from busy wait to condition variable based

### PR DESCRIPTION
**Pull request by s-sajid-ali**
Depends upon https://github.com/HEPonHPC/icaruswf/pull/28

Currently, we have a busy wait loop in `HepnosDataStore::wait()`. Typically, since we run `art` with multiple threads, this is the only thread among them that should be running, thereby freeing up resources for the detached `HEPnOS` thread to run. If the assumptions listed in the prior statement are incorrect, this busy wait loop might be causing slowdowns. Hence, this PR aims to fix that by removing the loop.

Currently testing this on `csresearch` and `theta` and will remove the `draft` status if it works on both platforms.

### CODE REVIEW

File: fcl/icarus_SH_queue.fcl.in [hunk 1]
```diff
@@ -59,6 +59,7 @@ outputs: {
       ]
       streamName: "out"
    }
+
```
Comment (marcpaterno) [inline fcl/icarus_SH_queue.fcl.in]: This change just adds whitespace. Let’s not make commits that do that except in a PR that only addresses formatting.

File: src/modules/mpi_wrapper.cc [hunk 2]
```diff
@@ -10,30 +12,47 @@
...
   try {
     TCLAP::CmdLine cmd(
-      "ICARUSWF with HEPnOS mpi-wrapper for launching art over MPI!");
+      "ICARUSWF with HEPnOS/ROOT mpi-wrapper for launching art over MPI!");
 
     TCLAP::ValueArg<int64_t> nevtsArg(
       "n",
       "num_evts_per_rank",
-     "total number of events to process, by each MPI rank",
+      "total number of events to process, by each MPI rank",
```
Comment (marcpaterno) [inline src/modules/mpi_wrapper.cc:55]: Is this the total number of event to process for the whole program? Or is the the number of events that each MPI rank will process? If the latter, what happens if the total number of events available to the input (whether that is a set of ROOT files or a HEPnOS store) is not an integer multiple of the number of MPI ranks?

Comment (s-sajid-ali) [reply src/modules/mpi_wrapper.cc:55]: It is the number of events that each MPI rank will process. Currently the ranks that get instructed to load missing events don't do anything so it is upon the user to ensure that the number of events for the given number of MPI ranks is appropriate. Perhaps we could add some error checking and warn the user of potential mistakes.

**Comment by marcpaterno** [comment]
We are merging this pull request with the intention of making substantial changes to `mpi_wrapper` afterward. We should review those changes together when they are complete.

**Pull request merged**